# footprinting_ATACseq.smk

Snakemake workflow for footprinting analysis of ATAC-seq data using [TOBIAS](https://github.com/loosolab/TOBIAS/).

!!! note

    Please make sure that you have [Singularity](https://sylabs.io/guides/3.7/user-guide/quick_start.html) and [Snakemake](https://snakemake.readthedocs.io/en/stable/) installed on your system and cloned the [SnakeNgs](https://github.com/NaotoKubota/SnakeNgs) repository.

## Workflow

<figure markdown="span">
	![footprinting_ATACseq.smk rulegraph](https://github.com/NaotoKubota/SnakeNgs/blob/develop/img/footprinting_ATACseq_rulegraph.svg?raw=true){ width="1000" align="center" }
</figure>

<span style="font-size: 0.8em; color: rgba(0, 0, 0, 0.4);">The rulegraph was created by [snakevision](https://github.com/OpenOmics/snakevision).</span>

1. Merge peak resions from all samples using [bedtools](https://bedtools.readthedocs.io/en/latest/) `merge`.
2. Merge bam files from all samples using [samtools](http://www.htslib.org/) `merge`.
3. Bias correction of ATAC-seq reads in open chromatin using [TOBIAS](https://github.com/loosolab/TOBIAS/) `ATACorrect`.
4. Calculate footprint scores from corrected cutsites using [TOBIAS](https://github.com/loosolab/TOBIAS/) `FootprintScores`.
5. Estimation of differentially bound motifs based on scores, sequence and motifs using [TOBIAS](https://github.com/loosolab/TOBIAS/) `BINDetect`.

## Usage

``` bash
snakemake -s /path/to/SnakeNgs/snakefile/footprinting_ATACseq.smk \
--configfile /path/to/config.yaml \
--cores <int> \
--use-singularity \
--rerun-incomplete
```

`config.yaml` should contain the following information:

``` yaml
# general
workdir: /path/to/output
experiment_table: /path/to/experiment_table.tsv

# TOBIAS
genome_fasta: /path/to/genome.fa
cluster_motifs: /path/to/motifs
```

- `path/to/output` is the directory where the output files will be saved.
- `path/to/experiment_table.tsv` is a tab-separated file containing the following information:

``` text
sample	bam	peak	group
SRRXXXXXX	/path/to/SRRXXXXXX.sort.rmdup.bam	/path/to/SRRXXXXXX_peaks.narrowPeak	Ref
SRRYYYYYY	/path/to/SRRYYYYYY.sort.rmdup.bam	/path/to/SRRYYYYYY_peaks.narrowPeak	Ref
SRRZZZZZZ	/path/to/SRRZZZZZZ.sort.rmdup.bam	/path/to/SRRZZZZZZ_peaks.narrowPeak	Ref
SRRAAAAAA	/path/to/SRRAAAAAA.sort.rmdup.bam	/path/to/SRRAAAAAA_peaks.narrowPeak	Alt
SRRBBBBBB	/path/to/SRRBBBBBB.sort.rmdup.bam	/path/to/SRRBBBBBB_peaks.narrowPeak	Alt
SRRCCCCCC	/path/to/SRRCCCCCC.sort.rmdup.bam	/path/to/SRRCCCCCC_peaks.narrowPeak	Alt
```

It is expected that the input BAM files are generated by the [preprocessing_ChIPseq.smk](preprocessing_ChIPseq.md) workflow and the peak files are generated by the [callpeak_ATACseq.smk](callpeak_ATACseq.md) workflow.

- `genome.fa` is the genome fasta file (e.g. `hg38.fa`).
- `motifs` is File containing motifs in either PFM, JASPAR or MEME format. These are the motifs which will be used to scan for binding sites.

## Docker image used in the workflow

- [quay.io/biocontainers/bedtools:2.24--1](https://quay.io/repository/biocontainers/bedtools)
- [quay.io/biocontainers/samtools:1.18--h50ea8bc_1](https://quay.io/repository/biocontainers/samtools)
- [quay.io/biocontainers/tobias:0.16.0--py38h24c8ff8_0](https://quay.io/repository/biocontainers/tobias)
